wa/workloads/jetnews/uiauto/app/src/main/java/com/arm/wa/uiauto/jetnews/UiAutomationJankTests.java

```diff
@@ -0,0 +1,298 @@
+// by workload-automation. We use an instance of UiAutomation so
+// things get setup properly (parameters, screen orientation etc).
+
+public class UiAutomationJankTests extends JankTestBase {
+    private static final int DEFAULT_BENCHMARK_REPEAT_COUNT = 1;
+    private static final int DEFAULT_TIMEOUT = 1000;
+    private static final int DEFAULT_BENCHMARK_FLING_SPEED = 5000;
+    private static final String[] DEFAULT_BENCHMARK_TESTS
+        = {"PortraitVerticalTest",
+           "PortraitHorizontalTest",
+           "LandscapeVerticalTest"};
+    private static final String PACKAGE_NAME = "com.example.jetnews";
+    private static final String LOG_TAG = "JetNewsJankTests: ";
+
+    private static final String MAIN_VIEW = "ArticlesMainScrollView";
+    private static final String TOP_ARTICLE = "TopStoriesForYou";
+    private static final String BOTTOM_ARTICLE = "PostCardHistory19";
+    private static final String POPULAR_LIST = "PopularOnJetnewsRow";
+    private static final String ARTICLE_VIEW = "ArticleHomeScreenPreview0";
+
+    private UiAutomation mUiAutomation;
+    private int repeat;
+    private int fling_speed;
+    private String[] tests;
+    private UiDevice mDevice;
+    private boolean testPortraitVertical;
+    private boolean testPortraitHorizontal;
+    private boolean testLandscapeVertical;
+
+    @JankTest(
+        expectedFrames = 100,
+        defaultIterationCount = 1
+    )
+
+    @GfxMonitor(processName = PACKAGE_NAME)
+    public void test1() throws Exception {
+        for (int i = 0; i < repeat; i++) {
+            if (testPortraitVertical) {
+                ActionLogger logger
+                    = new ActionLogger("PortraitVerticalTest",
+                                       mUiAutomation.getParams());
+                logger.start();
+                runPortraitVerticalTests();
+                logger.stop();
+            }
+
+            if (testPortraitHorizontal) {
+                ActionLogger logger
+                    = new ActionLogger("PortraitHorizontalTest",
+                                       mUiAutomation.getParams());
+                logger.start();
+                runPortraitHorizontalTests();
+                logger.stop();
+            }
+
+            if (testLandscapeVertical) {
+                ActionLogger logger
+                    = new ActionLogger("LandscapeVerticalTest",
+                                       mUiAutomation.getParams());
+                logger.start();
+                runLandscapeVerticalTests();
+                logger.stop();
+            }
+        }
+    }
+
+    private void runPortraitVerticalTests() throws Exception {
+        // Scroll to the first postcard in the list.
+        mDevice.wait(Until.findObject(By.res(MAIN_VIEW)), DEFAULT_TIMEOUT);
+        UiObject2 articles = mDevice.findObject(By.res(MAIN_VIEW));
+        ViewMatchers.assertThat(articles, CoreMatchers.notNullValue());
```

Comment:
Attempting to run this test on my emulator with the latest jetnews app results in the following crash.

```
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: class=com.arm.wa.uiauto.jetnews.UiAutomationJankTests
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: current=1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: id=AndroidJUnitRunner
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: numtests=1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: stream=
2024-09-12 17:29:36,628 ERROR        signal:         com.arm.wa.uiauto.jetnews.UiAutomationJankTests:
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: test=test1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS_CODE: 1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: class=com.arm.wa.uiauto.jetnews.UiAutomationJankTests
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: current=1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: id=AndroidJUnitRunner
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: numtests=1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: stack=java.lang.reflect.InvocationTargetException
2024-09-12 17:29:36,628 ERROR        signal:         	at java.lang.reflect.Method.invoke(Native Method)
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.jank.JankTestBase.runTest(JankTestBase.java:174)
2024-09-12 17:29:36,628 ERROR        signal:         	... 26 trimmed
2024-09-12 17:29:36,628 ERROR        signal:         Caused by: junit.framework.AssertionFailedError:
2024-09-12 17:29:36,628 ERROR        signal:         Expected: not null
2024-09-12 17:29:36,628 ERROR        signal:              Got: was null
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:16)
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:1)
2024-09-12 17:29:36,628 ERROR        signal:         	at com.arm.wa.uiauto.jetnews.UiAutomationJankTests.runPortraitVerticalTests(UiAutomationJankTests.java:116)
2024-09-12 17:29:36,628 ERROR        signal:         	at com.arm.wa.uiauto.jetnews.UiAutomationJankTests.test1(UiAutomationJankTests.java:88)
2024-09-12 17:29:36,628 ERROR        signal:         	... 29 more
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: stream=
2024-09-12 17:29:36,628 ERROR        signal:         Error in test1(com.arm.wa.uiauto.jetnews.UiAutomationJankTests):
2024-09-12 17:29:36,628 ERROR        signal:         java.lang.reflect.InvocationTargetException
2024-09-12 17:29:36,628 ERROR        signal:         	at java.lang.reflect.Method.invoke(Native Method)
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.jank.JankTestBase.runTest(JankTestBase.java:174)
2024-09-12 17:29:36,628 ERROR        signal:         	... 26 trimmed
2024-09-12 17:29:36,628 ERROR        signal:         Caused by: junit.framework.AssertionFailedError:
2024-09-12 17:29:36,628 ERROR        signal:         Expected: not null
2024-09-12 17:29:36,628 ERROR        signal:              Got: was null
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:16)
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:1)
2024-09-12 17:29:36,628 ERROR        signal:         	at com.arm.wa.uiauto.jetnews.UiAutomationJankTests.runPortraitVerticalTests(UiAutomationJankTests.java:116)
2024-09-12 17:29:36,628 ERROR        signal:         	at com.arm.wa.uiauto.jetnews.UiAutomationJankTests.test1(UiAutomationJankTests.java:88)
2024-09-12 17:29:36,628 ERROR        signal:         	... 29 more
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS: test=test1
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_STATUS_CODE: -2
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_RESULT: stream=
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         Time: 1.761
2024-09-12 17:29:36,628 ERROR        signal:         There was 1 failure:
2024-09-12 17:29:36,628 ERROR        signal:         1) test1(com.arm.wa.uiauto.jetnews.UiAutomationJankTests)
2024-09-12 17:29:36,628 ERROR        signal:         java.lang.reflect.InvocationTargetException
2024-09-12 17:29:36,628 ERROR        signal:         	at java.lang.reflect.Method.invoke(Native Method)
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.jank.JankTestBase.runTest(JankTestBase.java:174)
2024-09-12 17:29:36,628 ERROR        signal:         	... 26 trimmed
2024-09-12 17:29:36,628 ERROR        signal:         Caused by: junit.framework.AssertionFailedError:
2024-09-12 17:29:36,628 ERROR        signal:         Expected: not null
2024-09-12 17:29:36,628 ERROR        signal:              Got: was null
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:16)
2024-09-12 17:29:36,628 ERROR        signal:         	at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:1)
2024-09-12 17:29:36,628 ERROR        signal:         	at com.arm.wa.uiauto.jetnews.UiAutomationJankTests.runPortraitVerticalTests(UiAutomationJankTests.java:116)
2024-09-12 17:29:36,628 ERROR        signal:         	at com.arm.wa.uiauto.jetnews.UiAutomationJankTests.test1(UiAutomationJankTests.java:88)
2024-09-12 17:29:36,628 ERROR        signal:         	... 29 more
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         FAILURES!!!
2024-09-12 17:29:36,628 ERROR        signal:         Tests run: 1,  Failures: 1
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:
2024-09-12 17:29:36,628 ERROR        signal:         INSTRUMENTATION_CODE: -1
```

Do you see the same behaviour?

Reply:
Did you use the stock jetnews app? If so, it doesn't have the required code, so I'd expect it to fail. You need a modified version of JetNews with test tags and additional articles.

I have shared a page that contains both the apk and the repository that can be used to build it from sources.